### DISCUSSION TITLE
Giotto save and read

### DISCUSSION THREAD

**Discussion by Umaarasu** [giotto-suite/Giotto#737] [Q&A]
Hi,

I am curious about how everyone is saving and later reading the file. I was hoping that when I save the Giotto object, I can always read the object and start analysis from where I left. 

**For example I save the object after rank enrichment and I want to read the object and plot some charts. 

spatCellPlot2D(gobject = test,
               spat_enr_names = 'rank',
               cell_annotation_values = colnames(rank_matrix),
               cow_n_col = 2,coord_fix_ratio = 1, point_size = 1,
               save_param = list(save_name = "spat_enr_Rank_plot"))
 
Error in get_cell_metadata(gobject, spat_unit = spat_unit, feat_type = feat_type,  : 
  metadata referenced does not exist.** 

path_to_folder= /file/to/giottoobject/
test = loadGiotto(path_to_folder = path_to_folder))
test
An object of class giotto 
>Active spat_unit:  aggregate 
>Active feat_type:  rna 
[SUBCELLULAR INFO]
[AGGREGATE INFO]
expression -----------------------
  [cell][rna] raw normalized scaled
spatial locations ----------------
  [cell] raw
dim reduction --------------------
  [cell][rna] pca umap tsne
nearest neighbor networks --------
  [cell][rna] sNN.pca
attached images ------------------
giottoImage      : image

Could you please let me know how to fix this. Thanks!

**Comment by RubD** [comment]
Could you make a reproducible mini script? Saving and loading Giotto objects should work, see https://giottosuite.readthedocs.io/en/latest/subsections/datasets/230414_save_giotto_object_George.html for more details.

You can use one of the stored Giotto mini objects if that's easier, see `loadGiottoMini`

**Reply by Umaarasu** [reply]
Hi, I had been verifying the tutorial to save and load giotto objects. The stored Giotto mini object works fine when loading the object. It is my object I am having questions about. I am expecting that when I save a Giotto object it should technically store all the images and rank matrix that was generated along the way right? Does the 'save giotto' command work the same way as 'saveRDS' that we use for seurat object?

**Reply by rbutleriii** [reply]
Insofar as it saves the images, yes, though the output is a folder not an RDS file. See https://drieslab.github.io/Giotto_website/articles/saving_object.html
